lib/buy/onramper/onramper_buy_provider.dart

```diff
@@ -31,74 +43,330 @@ class OnRamperBuyProvider extends BuyProvider {
+    };
+
+    final url = Uri.https(_baseApiUrl, '$supported$paymentTypes/$fiatCurrency', params);
+
+    try {
+      final response =
+          await http.get(url, headers: {'Authorization': _apiKey, 'accept': 'application/json'});
+
+      if (response.statusCode == 200) {
+        final Map<String, dynamic> data = jsonDecode(response.body) as Map<String, dynamic>;
+        final List<dynamic> message = data['message'] as List<dynamic>;
+        return message
+            .map((item) => PaymentMethod.fromOnramperJson(item as Map<String, dynamic>))
+            .toList();
+      } else {
+        print('Failed to fetch available payment types');
+        return [];
+      }
+    } catch (e) {
+      print('Failed to fetch available payment types: $e');
+      return [];
+    }
+  }
+
+  Future<Map<String, dynamic>> getOnrampMetadata() async {
+    final url = Uri.https(_baseApiUrl, '$supported/onramps/all');
+
+    try {
+      final response =
+          await http.get(url, headers: {'Authorization': _apiKey, 'accept': 'application/json'});
+
+      if (response.statusCode == 200) {
+        final Map<String, dynamic> data = jsonDecode(response.body) as Map<String, dynamic>;
+
+        final List<dynamic> onramps = data['message'] as List<dynamic>;
+
+        final Map<String, dynamic> result = {
+          for (var onramp in onramps)
+            (onramp['id'] as String): {
+              'displayName': onramp['displayName'] as String,
+              'svg': onramp['icons']['svg'] as String
+            }
+        };
+
+        return result;
+      } else {
+        print('Failed to fetch onramp metadata');
+        return {};
+      }
+    } catch (e) {
+      print('Error occurred: $e');
+      return {};
     }
   }
 
-  String getColorStr(Color color) {
-    return color.value.toRadixString(16).replaceAll(RegExp(r'^ff'), "");
+  @override
+  Future<List<Quote>?> fetchQuote(
+      {required Currency sourceCurrency,
+      required Currency destinationCurrency,
+      required double amount,
+      required bool isBuyAction,
+      required String walletAddress,
+      PaymentType? paymentType,
+      String? countryCode}) async {
+    String? paymentMethod;
+
+    if (paymentType != null && paymentType != PaymentType.all) {
+      paymentMethod = normalizePaymentMethod(paymentType);
+      if (paymentMethod == null) paymentMethod = paymentType.name;
+    }
+
+    final actionType = isBuyAction ? 'buy' : 'sell';
+
+    final normalizedSourceCurrency = _getNormalizeCryptoCurrency(sourceCurrency);
+    final normalizedDestinationCurrency = _getNormalizeCryptoCurrency(destinationCurrency);
+
+    final params = {
+      'amount': amount.toString(),
+      if (paymentMethod != null) 'paymentMethod': paymentMethod,
+      'uuid': 'acad3928-556f-48a1-a478-4e2ec76700cd',
+      'clientName': 'CakeWallet',
+      'type': actionType,
+      'walletAddress': walletAddress,
+      'isRecurringPayment': 'false',
+      'input': 'source',
```

Comment:
whats `uuid` and `input`?

Reply:
uuid - Unique user id, I got it from the API docs. I’m not sure what it means, but they use CakeWallet as the default value. https://docs.onramper.com/reference/get_quotes-fiat-crypto 
input  - this like a fixed rate for exchange, I believe

Reply:
yup, reading the docs, input is like a fixed rate indeed
as an enhancement later on, we can make it like the exchange where if the user wants to buy a fixed crypto amount we can give him that option in the UI and change this to `destination` instead of `source`

for the uuid I will check with them whether we need it or not